miner/Miner.hs

```diff
@@ -263,7 +285,72 @@ scheme :: Env -> (TargetBytes -> HeaderBytes -> RIO Env HeaderBytes)
 scheme env = case envCmd env of
     CPU e _ -> cpu e
     GPU e _ -> gpu e
-    Keys -> error "Impossible: You shouldn't reach this case."
+    _ -> error "Impossible: You shouldn't reach this case."
+
+
+{- this will be ugly -}
+genLocal
+    :: ChainwebVersion
+    -> ChainId
+    -> P.Command Text
+    -> ClientM (P.CommandResult Hash)
+genLocal version cid =
+  case someChainwebVersionVal version of
+    SomeChainwebVersionT (_ :: Proxy cv) ->
+      case someChainIdVal cid of
+        SomeChainIdT (_ :: Proxy ccid) ->
+          client (Proxy :: Proxy (PactLocalApi cv ccid))
+
+getBalances :: BaseUrl -> Miner -> IO ()
+getBalances url (Miner (MinerId mi) _) = do
+    balanceStmts <- newManager (mkManagerSettings ss Nothing) >>= runClientM go . cenv
+    case balanceStmts of
+      Left e -> throwM e
+      Right bbs -> do
+        let f :: Decimal -> (Decimal, Text) -> IO Decimal
+            f tot (bal, bs) = do
+              printf (T.unpack bs)
+              return (tot + bal)
+        tot <- foldM f (0 :: Decimal) bbs
+        printf $ "Your total is " <> show (roundTo 12 tot) <> " kda.\n"
+  where
+    ss :: TLSSettings
+    ss = TLSSettingsSimple True True True
+    cenv m = ClientEnv m url Nothing
+    go = do
+      NodeInfo v _ cs _ <- client (Proxy @NodeInfoApi)
+      forM (sort cs) $ \cidtext -> do
+          c <- chainIdFromText cidtext
+          ks <- liftIO $ testSomeKeyPairs
+          let _code = printf "(coin.get-balance \"%s\")" (T.unpack mi)
+              meta = def { _pmChainId = P.ChainId cidtext }
+          cmd <- liftIO $ mkExec _code Null meta (NEL.toList ks) (Just $ P.NetworkId $ toText v) Nothing
+          (P.PactResult result) <- P._crResult <$> genLocal v c cmd
+          return $ case result of
+            Right (PLiteral (LDecimal bal)) ->
+              (bal, "The balance on chain " <> cidtext <> " is " <> (T.pack $ show bal) <> " kda.\n")
+            _ -> (0, "Something went wrong when looking for the balance on chain " <> cidtext <> "\n")
```

Comment:
capital `KDA` for currency codes  or better, use `₭`!

Reply:
Use `KDA`, not the Loas Kip symbol.